sync/sync.go

```diff
@@ -320,9 +322,13 @@ Register:
 		case action = <-replica.ActionChannel:
 		}
 	}
+
+	logrus.Infof("ACTION RECEIVED: %v", action)
 	if action == "start" {
 		logrus.Infof("Received start from controller")
 		return t.client.Start(replicaAddress)
+	} else if action == "done" {
```

Comment:
this is not required, after getting 200 OK from controller if replicaCount > 0, replica should continue and perform various tasks such as get added to controller, rebuild, sync etc.